packit_service/worker/handlers/abstract.py

```diff
@@ -161,12 +174,47 @@ class JobHandler(Handler):
     triggers: List[TheJobTriggerType]
 
     def __init__(
-        self, config: ServiceConfig, job_config: Optional[JobConfig], event: Event
+        self,
+        package_config: PackageConfig,
+        job_config: Optional[JobConfig],
+        event: dict,
     ):
-        super().__init__(config)
+        self.package_config: PackageConfig = package_config
         self.job_config: Optional[JobConfig] = job_config
+        self.event_type: str = event.get("event_type")
+        self.trigger: TheJobTriggerType = TheJobTriggerType(event.get("trigger"))
+        self.user_login: str = event.get("user_login")
+        self.trigger_id: int = event.get("trigger_id")
+        self.project_url: str = event.get("project_url")
+
         self.event = event
+        self._db_trigger: Optional[AbstractTriggerDbType] = None
+        self._project: Optional[GitProject] = None
+        self._config: Optional[ServiceConfig] = None
         self._clean_workplace()
 
+    @property
+    def db_trigger(self):
+        if not self._db_trigger and self.trigger_id is not None:
+            if self.trigger in (
+                TheJobTriggerType.pull_request,
+                TheJobTriggerType.pr_comment,
+                TheJobTriggerType.pr_label,
+            ):
+                self._db_trigger = PullRequestModel.get_by_id(self.trigger_id)
+            elif self.trigger == TheJobTriggerType.issue_comment:
+                self._db_trigger = IssueModel.get_by_id(self.trigger_id)
+            elif self.trigger == TheJobTriggerType.release:
+                self._db_trigger = ProjectReleaseModel.get_by_id(self.trigger_id)
+            elif self.trigger in TheJobTriggerType.push:
+                self._db_trigger = GitBranchModel.get_by_id(self.trigger_id)
+        return self._db_trigger
+
+    @property
+    def project(self) -> Optional[GitProject]:
+        if not self._project and self.project_url:
+            self._project = self.config.get_project(url=self.project_url)
```

Comment:
`self.config` is a `ServiceConfig`, right? And that one does not have `get_project()`. Or am I missing something?

Reply:
> self.config is a ServiceConfig, right?

yes

> And that one does not have get_project(). Or am I missing something?
 
ServiceConfig inherits from Config in packit and it has method get_project()

Reply:
Oh, okay, so that's what I missed :)